crates/autopilot/src/run_loop.rs

```diff
@@ -734,40 +752,35 @@ impl RunLoop {
         request: &solve::Request,
     ) -> Result<Vec<Result<competition::Solution, domain::competition::SolutionError>>, SolveError>
     {
+        let authenticator = self.eth.contracts().authenticator();
+        let is_allowed = authenticator
+            .is_solver(driver.submission_address.into())
+            .call()
+            .await;
+
+        // Do not send the request to the driver if the solver is denied
+        match is_allowed {
+            Ok(true) => {}
+            Ok(false) => return Err(SolveError::SolverDenyListed),
+            Err(err) => {
+                tracing::warn!(
+                    driver = driver.name,
+                    ?driver.submission_address,
+                    ?err,
+                    "failed to check if solver is deny listed"
+                );
+                return Err(SolveError::SolverDenyListed);
+            }
+        }
```

Comment:
This can be simplified. Or even using `and_then/flat_map`.
```suggestion
        let is_allowed = authenticator
            .is_solver(driver.submission_address.into())
            .call()
            .await
            .map_err(err {
              tracing::warn!(
                    driver = driver.name,
                    ?driver.submission_address,
                    ?err,
                    "failed to check if solver is deny listed"
                );
                SolveError::SolverDenyListed
            })?;
            
            if !is_allowed {
              return Err(SolveError::SolverDenyListed);
            }
```